### PR TITLE
moved buttons arround and added some fun style things :fireworks:

### PR DESCRIPTION
**Pull request by dylan-wright**
Live on dev.dcwright.xyz

### What it does
- moves around some buttons
- adds some read/write permissions to galleries (not :100:% enforced at the moment)
- fixes bugs in gallery controller
- fun style things (check out the thumbnail selection feature in the gallery edit page)

### CODE REVIEW

File: modules/gallery/client/components/gallery-detail.component.ts [hunk 1]
```diff
@@ -4,17 +4,21 @@ import { Router, ActivatedRoute } from '@angular/router';
...
 
+import { AuthService } from '../../../auth/client/services/auth.service';
+
 @Component({
   templateUrl: '../views/gallery-detail.view.html'
 })
 export class GalleryDetailComponent {
   private gallery: Gallery;
   private endpoint: string;
+  private user: any;
```
Comment (JasonSHD) [inline modules/gallery/client/components/gallery-detail.component.ts:15]: Let's use a typed language.... no wait, let's not use types.... 😜

File: modules/gallery/client/components/gallery-detail.component.ts [hunk 2]
```diff
@@ -23,9 +27,19 @@ export class GalleryDetailComponent {
...
+    this.user = this.authService.getUser();
   }
   
   uriChange(data: any) {
     this.gallery = data.gallery;
   }
+
+  isAllowedToEdit() {
+    if (this.gallery) {
+      return this.gallery.publicWrite || this.user._id === this.gallery.creator || this.gallery.allowedUsers.includes(this.user._id);
```
Comment (JasonSHD) [inline modules/gallery/client/components/gallery-detail.component.ts:40]: Lines that get this long should probably be multi-line statements.  This is kinda right at the edge, so we can keep it the way it is.

File: modules/gallery/client/models/gallery.model.ts [hunk 3]
```diff
@@ -1,7 +1,33 @@
...
+  thumbnail: string;
+  publicRead: boolean;
+  publicWrite: boolean;
+  creator: string;
+  allowedUsers: Array<string>;
+
   constructor(
-    { id='', name='', pictures=[], thumbnail=''}:
-    { id?: string; name?: string; pictures?: Array<any>; thumbnail?: string }={}
+    {
```
Comment (JasonSHD) [inline modules/gallery/client/models/gallery.model.ts:12]: If you want to fix

File: modules/gallery/client/views/gallery-thumbnails.view.html [hunk 4]
```diff
@@ -1,16 +1,20 @@
 <div class="card-deck">
-  <div class="card" *ngFor="let gallery of galleries">
+  <div class="card" *ngFor="let gallery of galleries"
```
Comment (JasonSHD) [inline modules/gallery/client/views/gallery-thumbnails.view.html:2]: I highly approve.  They should be nested, but this is close enough.

File: modules/gallery/server/controllers/gallery.controller.js [hunk 5]
```diff
@@ -326,12 +325,29 @@ function galleryController(logger, shared) {
...
    * @param {string}  resource  The resource being requested.
+   * @param {string}  operation - whether the resource is being read or written
    *
    * @return {boolean}
    */
-  function isAuthorized(user, resource) {
-    // TODO: Currently set to deny any request not approved by the resource manager.
-    return false;
+  function isAuthorized(user, resource, operation) {
+    let authorized = false;
```
Comment (JasonSHD) [inline modules/gallery/server/controllers/gallery.controller.js:333]: Should be var 😆

**Review by JasonSHD — APPROVED** [review]
Looks good

File: modules/gallery/client/components/gallery-detail.component.ts (hunk 2 above)
Comment (dylan-wright) [reply modules/gallery/client/components/gallery-detail.component.ts:40]: Yeah I was being lazy

File: modules/gallery/server/controllers/gallery.controller.js (hunk 5 above)
Comment (dylan-wright) [reply modules/gallery/server/controllers/gallery.controller.js:333]: You monster

**Pull request merged**